schema.go

```diff
@@ -498,6 +506,44 @@ func (s *RecordSchema) String() string {
 	return `{"name":"` + s.FullName() + `","type":"` + typ + `","fields":[` + fields + `]}`
 }
 
+// Resolve returns the resolved form of the schema.
+func (s *RecordSchema) Resolve(cache *SchemaCache) string {
+	typ := "record"
+	if s.isError {
+		typ = "error"
+	}
+
+	fields := ""
+	for _, f := range s.fields {
+		switch lookup := f.typ.(type) {
+		case *RefSchema:
+			switch found := lookup.actual.(type) {
+			case *RecordSchema:
+				if strings.Contains(found.full, ".") {
+					dependency := cache.Get(found.full)
+					if dependency != nil {
+						fields += `{"name":"` + found.Name() + `","type":{` + dependency.Resolve(cache) + `}},`
+						fmt.Println(fields)
+					} else {
+						fields += found.Resolve(cache) + ","
+					}
+				} else {
+					fields += f.String() + ","
+				}
+			default:
+				fields += f.String() + ","
+			}
+		default:
+			fields += f.String() + ","
+		}
```

Comment:
This entire section can be done by:
`fields += f.Resolve() + ","`
If you make  `RefSchema.Resolve` returns the schema string instead of the full name.